commit 946c076f7fb77da5a55637036a706d1cf8767354
Author: Ladislav CABELKA <lcabelka@mandriva.com>
Date:   Fri Jun 20 13:16:48 2014 +0200

    pkg-server: do not pre-select menu if entry not exists

diff --git a/pulse2/services/pulse2/package_server/imaging/pxe/tracking.py b/pulse2/services/pulse2/package_server/imaging/pxe/tracking.py
index 60df2dd..7bd8e02 100644
--- a/pulse2/services/pulse2/package_server/imaging/pxe/tracking.py
+++ b/pulse2/services/pulse2/package_server/imaging/pxe/tracking.py
@@ -1,11 +1,11 @@
-#!/usr/bin/python 
+#!/usr/bin/python
 # -*- coding: utf-8; -*-
 #
 # (c) 2013 Mandriva, http://www.mandriva.com/
 #
 # This file is part of Pulse 2, http://pulse2.mandriva.org
 #
-# Pulse 2 is free software; you can redistribute it and/or modify  
+# Pulse 2 is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 2 of the License, or
 # (at your option) any later version.
@@ -29,7 +29,7 @@ class TrackingContainer(Singleton):
     entries = {}
 
     def add(self, mac, entry):
-        """ 
+        """
         Add the menu entry
 
         @param mac: MAC address of client machine
@@ -41,7 +41,7 @@ class TrackingContainer(Singleton):
         self.entries[mac] = entry, False
 
     def get(self, mac):
-        """ 
+        """
         Get the menu entry
 
         @param mac: MAC address of client machine
@@ -63,10 +63,10 @@ class TrackingContainer(Singleton):
         if mac in self.entries :
             entry, flag = self.entries[mac]
             self.entries[mac] = entry, True
-  
+
     def delete(self, mac):
-        """ 
-        Delete the menu entry 
+        """
+        Delete the menu entry
 
         @param mac: MAC address of client machine
         @type mac: str
@@ -74,6 +74,9 @@ class TrackingContainer(Singleton):
         if mac in self.entries :
             del self.entries[mac]
 
+    def __contains__(self, mac):
+        return mac in self.entries
+
 CHOOSEN_MENU_ENTRY = LOG_ACTION[1][1]
 
 class EntryTracking(TrackingContainer):
@@ -83,11 +86,11 @@ class EntryTracking(TrackingContainer):
     Second parameter confirms that menu entry is a choice of backup
     or restore.
 
-    The entries are added by extracting of log message from logCientAction, 
+    The entries are added by extracting of log message from logCientAction,
     which reports all client activities.
 
     """
- 
+
     def search_and_extract(self, mac, phase, message):
         """
         Log message extract
@@ -97,11 +100,11 @@ class EntryTracking(TrackingContainer):
 
         @param phase: step of imaging workflow
         @type phase: str
- 
+
         @param message: displayed message
         @type message: str
         """
-        if phase == LOG_STATE.MENU and CHOOSEN_MENU_ENTRY in message: 
+        if phase == LOG_STATE.MENU and CHOOSEN_MENU_ENTRY in message:
             entry = message.replace(CHOOSEN_MENU_ENTRY, "").replace(":","").strip()
             if entry.isdigit() :
                 num = int(entry)
